concordium-contracts-common-derive/src/derive.rs

```diff
@@ -279,24 +326,145 @@ impl ContainerAttributes {
         }
         bounds
     }
+
+    /// Get the repr attribute if set otherwise the smallest size length which
+    /// can represent the number of variants.
+    fn tag_repr(&self, variants_len: usize) -> syn::Result<syn::Ident> {
+        if let Some(repr_attribute) = &self.repr {
+            let max = repr_attribute.max_variant_length();
+            let ident = repr_attribute.ident();
+            ensure!(
+                max >= variants_len,
+                Span::call_site(),
+                "Too many variants. Maximum {max} are supported with 'repr({ident})'",
+            );
+            Ok(ident)
+        } else if variants_len <= usize::from(u8::MAX) + 1 {
+            Ok(format_ident!("u8"))
+        } else if variants_len <= usize::from(u16::MAX) + 1 {
+            Ok(format_ident!("u16"))
+        } else {
+            bail!(
+                Span::call_site(),
+                "[derive(Deserial)]: Too many variants. Maximum 65536 are supported.",
```

Comment:
Is that the right error? This function is also called from `Serial`, and there are cases where you want to only derive that.

Reply:
right, I'll remove the mentioning of Deserial